src/MCPClient/lib/clientScripts/removeUnneededFiles.py

```diff
@@ -23,23 +25,29 @@ def remove_file(target_file, file_uuid):
     except ConfigParser.NoOptionError:
         removableFiles = ["Thumbs.db", "Icon", u"Icon\u000D", ".DS_Store"]
 
-    basename = os.path.basename(target_file)
-    if basename in removableFiles:
-        print("Removing {filename} (UUID: {uuid})".format(uuid=file_uuid, filename=basename))
-        try:
-            os.remove(target_file)
-        except OSError:
-            shutil.rmtree(target_file)
-        # Gearman passes parameters as strings, so None (NoneType) becomes
-        # "None" (string)
-        if file_uuid and file_uuid != "None":
-            fileWasRemoved(file_uuid)
-
+    ### 1. Remove unneeded files from package.
+    for dir_path, dir_name, filenames in os.walk(sip_dir):
+        for filename in filenames:
+            if filename in removableFiles:
+                del_file = os.path.join(dir_path, filename)
+                try:
+                    os.remove(del_file)
+                except OSError:
+                    shutil.rmtree(del_file)
+                logger.debug('removeUnneededFiles file deleted: %s', del_file)
+
+    ### 2. Insert removal notification into events database
+    sip_files = File.objects.filter(sip=sip_UUID).values_list('uuid', 'currentlocation')
+    remove_files = [file_UUID for file_UUID, current_location in sip_files if (os.path.basename(current_location) in removableFiles and file_UUID != 'None')]
```

Comment:
This list comprehension seems to have a bug. I created a zip file containing a file name `Icon` and ran it though. The file was correctly deleted, but wasn't deleted in the DB.  Can you look into this?

To reproduce:
* Use `archivematica-sampledata/TestTransfers/mac-Icon`
* Create a zip containing the icon & something else `zip has-icon.zip Icon MARBLES.GIF`
* Run though Archivematica
* The file is deleted, but the entry is still in the DB and has no removal time

Also, the logger doesn't write anything below info level, so you may want to change them to use `logger.info`.  I think this is useful enough information to print every time.